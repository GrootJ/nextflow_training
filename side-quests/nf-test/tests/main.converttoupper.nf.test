nextflow_process {

    name "Test Process convertToUpper"
    script "main.nf"
    process "convertToUpper"

    test("Should run without failures and with correct outputs") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = file("test-file.txt")
                // input[0] = file("Hello-output.txt")
                input[0] = "${projectDir}/greetings.csv"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
