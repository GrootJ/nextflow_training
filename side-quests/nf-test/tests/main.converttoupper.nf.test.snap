{
    "Should run without failures and with correct outputs": {
        "content": [
            {
                "0": [
                    "UPPER-greetings.csv:md5,f36624b0e040fa880e61fb1304b4b6b9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-02T22:49:00.099117"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "UPPER-greetings.csv:md5,f36624b0e040fa880e61fb1304b4b6b9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-02T22:45:21.376548"
    }
}